src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs

```diff
@@ -79,44 +100,85 @@ public override Task SendGroupsAsync(IReadOnlyList<string> groupNames, string me
             throw new NotImplementedException();
         }
 
-        public override Task SendUserAsync(string userId, string methodName, object[] args, CancellationToken cancellationToken = default)
+        public override async Task SendUserAsync(string userId, string methodName, object[] args, CancellationToken cancellationToken = default)
         {
             var api = _restApiProvider.GetSendToUserEndpoint(userId);
-            return CallRestApi(api, HttpMethod.Post, methodName, args, cancellationToken);
+
+            var response = await CallRestApi(api, HttpMethod.Post, methodName, args, cancellationToken);
+
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (Exception ex)
+            {
+                ThrowException(ex, response.StatusCode, api.Audience);
+            }
         }
 
         public override Task SendUsersAsync(IReadOnlyList<string> userIds, string methodName, object[] args, CancellationToken cancellationToken = default)
         {
             throw new NotImplementedException();
         }
 
-        public Task UserAddToGroupAsync(string userId, string groupName, CancellationToken cancellationToken = default)
+        public async Task UserAddToGroupAsync(string userId, string groupName, CancellationToken cancellationToken = default)
         {
             var api = _restApiProvider.GetUserGroupManagementEndpoint(userId, groupName);
-            return CallRestApi(api, HttpMethod.Put, null, null, cancellationToken);
+
+            var response = await CallRestApi(api, HttpMethod.Put, null, null, cancellationToken);
+
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (Exception ex)
+            {
+                ThrowException(ex, response.StatusCode, api.Audience);
+            }
         }
 
-        public Task UserRemoveFromGroupAsync(string userId, string groupName, CancellationToken cancellationToken = default)
+        public async Task UserRemoveFromGroupAsync(string userId, string groupName, CancellationToken cancellationToken = default)
         {
             var api = _restApiProvider.GetUserGroupManagementEndpoint(userId, groupName);
-            return CallRestApi(api, HttpMethod.Delete, null, null, cancellationToken);
-        }
+            var response = await CallRestApi(api, HttpMethod.Delete, null, null, cancellationToken);
```

Comment:
>CallRestApi [](start = 33, length = 11)

move into try..catch block? and use `if (response.IsSuccessStatusCode)` instead?

Reply:
But I need exception thrown by `EnsureSuccessStatusCode`

Reply:
Do you mean : 
``` C#
            RestApiEndpoint api = null;
            HttpResponseMessage response = null;

            try
            {
                api = _restApiProvider.GetUserGroupManagementEndpoint(userId, groupName);
                response = await CallRestApi(api, HttpMethod.Delete, null, null, cancellationToken);
                response.EnsureSuccessStatusCode();
            }
            catch(Exception ex)
            {
                if (api != null && response != null)
                {
                    ThrowException(ex, response.StatusCode, api.Audience);
                }
                else
                {
                    throw;
                }
            }
```
?

But I don't get the point. I just want to set the HttpRequestException (if response status is failed) to the inner exception of my exception. I don't care about other exceptions.